lib/projfs.c

```diff
@@ -49,12 +50,21 @@
 
 struct projfs_config {
 	int initial;
+	int log;
+	char *log_path;
 };
 
 #define PROJFS_OPT(t, p, v) { t, offsetof(struct projfs_config, p), v }
 
 static struct fuse_opt projfs_opts[] = {
 	PROJFS_OPT("initial",	initial, 1),
+	PROJFS_OPT("--initial",	initial, 1),
+
+	PROJFS_OPT("log=",	log, 1),
+	PROJFS_OPT("log=%s",	log_path, 0),
+	PROJFS_OPT("--log=",	log, 1),
+	PROJFS_OPT("--log=%s",	log_path, 0),
```

Comment:
I understand why we need both `initial` and `--initial`, or `log=%s` and `--log=%s`, but what do we gain from having both the `log=` and `log=%s` variant? (Or `--log=` and `--log=%s`.)

Reply:
This is a trick [borrowed from libfuse](https://github.com/libfuse/libfuse/blob/f6691d0a20dc3c4434f48d885fa2b67d1ed7137a/lib/fuse.c#L4607) -- they both match an command-line argument like `--log=foo.log`, but the `--log=` match causes `config->log` to be set to `1`, while the actual value of the argument ends up in `config->log_path`.

But since you mention it, I see that that's probably done in libfuse because they are dealing with  options where the user-supplied value might be zero (`--gid=0`) and so just relying on a single option field isn't sufficient to determine if the option was set or not.  But we don't have that situation here, so I think you're right, I can remove the extra field and just check if `log_path` is `NULL`.  Thanks!!  👍